keras_cv/layers/preprocessing/random_cutout.py

```diff
@@ -162,17 +160,14 @@ def _parse_bounds(self, factor):
         else:
             return type(factor)(0), factor
 
-    def call(self, inputs, training=True):
-        if training is None:
-            training = backend.learning_phase()
+    def get_random_transformation(self):
+        return self._sample_num_cutouts()
 
-        augment = lambda: self._random_cutout(inputs)
-        no_augment = lambda: inputs
-        return tf.cond(tf.cast(training, tf.bool), augment, no_augment)
-
-    def _random_cutout(self, inputs):
+    @tf.function
```

Comment:
@qlzh727 I suppose that the model is already wrapped by default in a `tf.function` at build time (later we need to  identify how we want to control to activate `jit_compile` on a specific region/function)

Reply:
yea, I don't think the tf.function is necessary here.

Reply:
yeah we only decorate components with tf.function if we intend for them to be used in isolation AND they are computationally expensive.

i.e. coco metrics - they will be used outside of models most likely due to the complexity of BBOX models and they are extremely expensive so best to compile them.  Other than rare cases no need to decorate code with tf.function

Reply:
I am not sure why but if the function is not decorated with `tf.function`, the graph doesn't support the loop.
This problem persists when I create a one-layer model that applies CutOut:

```
OperatorNotAllowedInGraphError: Exception encountered when calling layer "random_cutout" (type RandomCutout).

Iterating over a symbolic `tf.Tensor` is not allowed in Graph execution. Use Eager execution or decorate this function with @tf.function.

Call arguments received by layer "random_cutout" (type RandomCutout):
  • inputs=tf.Tensor(shape=(None, 504, 672, 3), dtype=float32)
  • training=True
```

Reply:
I am working on this at:
https://github.com/tensorflow/tensorflow/pull/55350

I hope It could cover also this case.

Reply:
No point in merging this PR without the decorator as long as that isn't covered, right?

Reply:
no, we should add a test case to cover this as well.  Will you add a test case running cutout in a model?

Reply:
Sure, I will add the one-layer model in the test case!

Reply:
@LukeWood This hasn't been done yet, has it? I can create a pull request for it. 
I wonder if these model testcases would be useful for all layers.